packages/react/src/components/Autofill/Autofill.test.tsx

```diff
@@ -1,311 +1,369 @@
-    component = mount(
+
+    render(
       <Autofill
-        componentRef={autofillRef}
+        componentRef={autofillRef as IRefObject<IAutofill>}
         // eslint-disable-next-line @typescript-eslint/no-deprecated
         onInputChange={onInputChange}
         suggestedDisplayValue="he"
       />,
     );
 
-    autofill.inputElement!.value = 'hel';
-    ReactTestUtils.Simulate.input(autofill.inputElement!);
-    expect(autofill.value).toBe('hel');
+    const input = screen.getByRole('textbox') as HTMLInputElement;
 
-    ReactTestUtils.Simulate.compositionStart(autofill.inputElement!, {});
+    // initial
+    fireEvent.input(input, { target: { value: 'hel' } });
+    expect(autofillRef.current!.value).toBe('hel');
 
-    ReactTestUtils.Simulate.keyDown(autofill.inputElement!, {
+    fireEvent.compositionStart(input);
+
+    // "p" with isComposing true
+    fireEvent.keyDown(input, {
       keyCode: KeyCodes.p,
       which: KeyCodes.p,
+      isComposing: true,
     });
-    autofill.inputElement!.value = 'help';
-    ReactTestUtils.Simulate.input(autofill.inputElement!, {
-      target: autofill.inputElement!,
-      nativeEvent: {
-        isComposing: true,
-      } as any,
+
+    fireEvent.input(input, {
+      target: { value: 'help' },
+      isComposing: true,
     });
 
-    ReactTestUtils.Simulate.compositionEnd(autofill.inputElement!, {});
-    autofill.inputElement!.value = '🆘';
-    ReactTestUtils.Simulate.input(autofill.inputElement!, {
-      target: autofill.inputElement!,
-      nativeEvent: {
-        isComposing: true,
-      } as any,
+    // compositionEnd
+    fireEvent.compositionEnd(input);
+
+    fireEvent.input(input, {
+      target: { value: '🆘' },
+      isComposing: true,
     });
+
     jest.runOnlyPendingTimers();
 
-    ReactTestUtils.Simulate.keyDown(autofill.inputElement!, {
+    // type "m" while composing
+    fireEvent.keyDown(input, {
       keyCode: KeyCodes.m,
       which: KeyCodes.m,
-      nativeEvent: {
-        isComposing: true,
-      } as any,
+      isComposing: true,
     });
-    autofill.inputElement!.value = '🆘m';
-    ReactTestUtils.Simulate.input(autofill.inputElement!, {
-      target: autofill.inputElement!,
-      nativeEvent: {
-        isComposing: true,
-      } as any,
+
+    fireEvent.input(input, {
+      target: { value: '🆘m' },
+      isComposing: true,
     });
 
-    ReactTestUtils.Simulate.compositionEnd(autofill.inputElement!, {});
-    autofill.inputElement!.value = '🆘Ⓜ';
-    ReactTestUtils.Simulate.input(autofill.inputElement!, {
-      target: autofill.inputElement!,
-      nativeEvent: {
-        isComposing: false,
-      } as any,
+    fireEvent.compositionEnd(input);
+
+    fireEvent.input(input, {
+      target: { value: '🆘Ⓜ' },
+      isComposing: false,
     });
+
     jest.runOnlyPendingTimers();
 
     expect(onInputChange.mock.calls).toEqual([
       ['hel', false],
       ['help', true],
-      ['🆘', true], // from input event
```

Comment:
can we keep these comments? as asserting against the mock calls makes the assertion hard to understand